src/lightning/data/streaming/dataset.py

```diff
@@ -26,16 +27,25 @@
 from lightning.data.streaming.shuffle import FullShuffle, NoShuffle, Shuffle
 from lightning.data.utilities.env import Environment, _DistributedEnv, _WorkerEnv
 
+
+@dataclass
+class RemoteDir:
```

Comment:
Let's define this at the bottom. It's nicer to have the core api definition (streaming dataset) at the top of the file. Then you'll need to use string references, e.g. `input_dir: Union[str, "RemoteDir"],` for types.

Reply:
Sure.